log/wrapper.go

```diff
@@ -7,20 +7,53 @@ import (
 	sentry "github.com/getsentry/sentry-go"
 )
 
-// Wrapper is a simple wrapper of a logging function.
+// Wrapper defines a simple interface to wrap logging functions.
 //
+// As principles, library code should
+//
+// 1. Not do any logging, the library code should communicate errors back to the
+//    caller, and let the caller decide how to deal with them (log them, ignore
+//    them, panic, etc.)
+//
+// 2. In some rare cases, 1 is not possible, for example the error might happen
+//    in a background goroutine. In those cases some logging is necessary, but
+//    those should be kept at minimal, and the library code should provide
+//    control to the caller on how to do those logging.
+//
+// This interface is meant to solve Principle 2 above.
 // In reality we might actually use different logging libraries in different
 // services, and they are not always compatible with each other.
 // Wrapper is a simple common ground that it's easy to wrap whatever logging
 // library we use into.
 //
-// This is also the same type as thrift.Logger and can be used interchangeably
-// (sometimes with a typecast).
+// For production code,
```

Comment:
Should we also call out that this is really only meant for passing into libraries, production code can just use the functions provided by the `log` module?

Reply:
Done. PTAL?